src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -102,6 +102,148 @@ private PatternSyntax ParsePattern()
             return _syntaxFactory.ConstantPattern((ExpressionSyntax)node);
         }
 
+        //
+        // Parse an expression where a declaration expression would be permitted. This is suitable for use after
+        // the `out` keyword in an argument list, or in the elements of a tuple literal (because they may
+        // be on the left-hand-side of a deconstruction). The first element of a tuple is handled slightly
+        // differently, as we check for the comma before concluding that the identifier should cause a
+        // disambiguation. For example, for the input `(A < B , C > D)`, we treat this as a tuple with
+        // two elements, because if we considered the `A<B,C>` to be a type, it wouldn't be a tuple at
+        // all. Since we don't have such a thing as a one-element tuple (even for deconstruction), the
+        // absence of the comma after the `D` means we don't treat the `D` as contributing to the
+        // disambiguation of the expression/type. More formally, ...
+        //
+        // If a sequence of tokens can be parsed(in context) as a* simple-name* (§7.6.3), *member-access* (§7.6.5),
```

Comment:
That is what is in the draft spec. I agree it could be improved ever so slightly. If this is what is holding up your approval of the code review, I think we should hold this set of changes until I am back in the office in January.

---
In reply to: [91570733](https://github.com/dotnet/roslyn/pull/15548#discussion_r91570733) [](ancestors = 91570733,91453230,91392143)